sycl/test-e2e/ESIMD/api/functional/common.hpp

```diff
@@ -33,9 +33,10 @@ namespace details {
 template <typename T> bool are_bitwise_equal(T lhs, T rhs) {
   constexpr size_t size{sizeof(T)};
 
-  // Such type-punning is OK from the point of strict aliasing rules
-  const auto &lhs_bytes = reinterpret_cast<const unsigned char(&)[size]>(lhs);
-  const auto &rhs_bytes = reinterpret_cast<const unsigned char(&)[size]>(rhs);
+  std::array<char, size> lhs_bytes;
+  std::array<char, size> rhs_bytes;
+  std::memcpy(lhs_bytes.data(), &lhs, size);
```

Comment:
FYI, reinterpreting to (char*) should work and would avoid memcpy.

Reply:
I hit some problems with some of the different template types that call this function not being supported by reinterpret_cast so I went with this.

Reply:
I see, thanks for clarification.